src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -43,6 +43,11 @@ public class Settings : ISettings
 
         private readonly Dictionary<string, VirtualSettingSection> _computedSections;
 
+        public Dictionary<string, VirtualSettingSection> GetComputedSections()
```

Comment:
I was getting an error when I tried to make `_computedSections` public, so I created a property called `ComputedSections` instead that gets the value of `_computedSections`. 

The only thing I'm unsure about now is that I had to cast the ISettings object I was working with in `ConfigRunner.cs` to a Settings object to be able to use the new property. I don't know if this might cause any problems. The method I'm using to get the settings is in the same file as `_computedSections`, and so far I haven't encountered any issues, so it looks like it's fine?

Reply:
+1 to `ComputedSections`. I think it's good too, you're getting the settings from a function that is only defined in the `Settings` class https://github.com/NuGet/NuGet.Client/pull/5126/files#diff-3783860f3f9dc2e36126a8a162f70f9ab66956adab808551c91ebbf8a224b63aR48.

Reply:
> I was getting an error when I tried to make _computedSections public,

Right, you don't want to do this (that's a field, not a property)

> so I created a property called ComputedSections instead that gets the value of _computedSections

👍.  Also, reiterating Andy's comment, make sure you use an autoprop.

> The only thing I'm unsure about now is that I had to cast the ISettings object I was working with in ConfigRunner.cs to a Settings object to be able to use the new property

That's because the new property is only defined in that `Settings` class, and not the `ISettings`. That's a design decision for you to make. The cost of the current implementation is that it requires the `Settings` type, and all our other implementations (eg, `ImmutableSettings`, `NullSettings`) will not be aware of `ComputedSections`. If you're good with that, then the API change you've proposed makes sense to me.
I'd be careful for the case when there's only invalid nuget.configs: that looks like it would result in your code encountering `ImmutableSettings`, which your cast will fail. See: https://github.com/NuGet/NuGet.Client/blob/d0bb0a180105de35a29ee89f52282a0eabd20588/src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs#L511